draft-yasskin-dispatch-web-packaging.md

```diff
@@ -0,0 +1,650 @@
+    ones that drove HPACK, so we may be able to do better with another
+    compression format.
+
+The optional `length` field in the index entries is redundant with the length
+prefixes on the `response-headers` and `body` in the content, but it can be used
+to issue Range requests {{?RFC7233}} for responses that appear late in the
+content.
+
+
+## Parsing the manifest {#manifest}
+
+A package's manifest contains some metadata for the package; hashes, used in
+{{hashing-resources}}, for all resources included in that package; and
+validity information for any [sub-packages](#subpackages) the package depends
+on. The manifest is signed, so that UAs can trust that it comes from its claimed
+origin. [^ednote-manifest-name]{: source="jyasskin"}
+
+[^ednote-manifest-name]: This section doesn't describe a manifest
+    (https://www.merriam-webster.com/dictionary/manifest#h3), so consider
+    renaming it to something like "authenticity".
+
+To parse a manifest starting at *manifest-start*, a parser MUST do the following:
+
+Load one CBOR item starting at *manifest-start* as a `signed-manifest` from the
+following CDDL:
+
+~~~~~ cddl
+signed-manifest = {
+  manifest: manifest,
+  certificates: [+ certificate],
+  signatures: [+ signature]
+}
+
+manifest = {
+  metadata: manifest-metadata,
+  resource-hashes: {* hash-algorithm => [hash-value]},
+  ? subpackages: [* subpackage],
+}
+
+manifest-metadata = {
+  date: time,
+  origin: uri,
+  * tstr => any,
+}
+
+; From https://www.w3.org/TR/CSP3/#grammardef-hash-algorithm.
+hash-algorithm /= "sha256" / "sha384" / "sha512"
+; Note that a hash value is not base64-encoded, unlike in CSP.
+hash-value = bstr
+
+; X.509 format; see https://tools.ietf.org/html/rfc5280
+certificate = bstr
+
+signature = {
+  ; This is the index of the certificate within the
+  ; certificates array to use to validate the signature.
+  keyIndex: uint,
+  signature: bstr,
+}
+~~~~~
+
+If the item doesn't match the CDDL or it's not a Canonical CBOR item (Section
+3.9 of {{CBOR}}), parsing MUST fail.
+
+Parse the elements of `certificates` as X.509 certificates within the
+{{!RFC5280}} profile. If any certificate fails to parse, parsing MUST fail.
+
+Let *message* be the concatenation of the following byte strings. This matches
+the {{?TLS1.3=I-D.ietf-tls-tls13}} format to avoid cross-protocol attacks when
+TLS certificates are used to sign manifests.
+
+1. A string that consists of octet 32 (0x20) repeated 64 times.
+1. A context string: the ASCII encoding of "Web Package Manifest".
+1. A single 0 byte which serves as a separator.
+1. The bytes of the `manifest` CBOR item.
+
+Let *signing-certificates* be an empty array.
+
+For each element *signature* of `signatures`:
+
+1. Let *certificate* be `certificates`\[*signature*\["keyIndex"]].
+
+1. The parser MUST define a partial function from public key types to signing
+   algorithms, with the following map as a subset:
```

Comment:
"as a subset" isn't very clear to me, perhaps this is mathematician lingo though?